integration/test-mb/test-amqp/src/test/java/org/wso2/ei/mb/test/amqp/TopicBasicsTestCase.java

```diff
@@ -0,0 +1,162 @@
+            sender.sendMessages(1, "test message");
+
+            Assert.assertNotNull(subscriber.receiveMessage(1000L), "Topic not received by the subscriber");
+            Assert.assertNotNull(subscriber2.receiveMessage(1000L), "Topic not received by the 2nd subscriber");
+
+            //send 500 messages
+            subscriber.registerMessageListener();
+            subscriber2.registerMessageListener();
+            sender.sendMessages(500, "test message");
+            TimeUnit.SECONDS.sleep(5L);
+            Assert.assertEquals(subscriber.receivedMessageCount(), 500, "Did not receive all the topics");
+            Assert.assertEquals(subscriber2.receivedMessageCount(), 500, "Did not receive all the topics");
+        } finally {
+            if (sender != null) {
+                sender.closeSender();
+                subscriber.closeReceiver();
+            }
+        }
+    }
+
+    /**
+     * Test for durable subscription.
+     * 1. Create two durable subscribers.
+     * 2. Send 10 messages and check for their reception.
+     * 3. Close one subscriber.
+     * 4. Send 10 messages.
+     * 5. Start the closed subscriber again.
+     * 6. Check the reception of messages.
+     * @throws JMSException
+     * @throws NamingException
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    @Test(groups = "wso2.mb.topic")
+    public void testDurableSubscription() throws JMSException, NamingException, IOException, InterruptedException {
+        TopicSender sender = null;
+        TopicReceiver subscriber = null;
+        TopicReceiver subscriber2 = null;
+        try {
+            subscriber = new TopicReceiver("testTopic2", JMSAcknowledgeMode.AUTO_ACKNOWLEDGE, configurationReader,
+                    "sub1");
+
+            subscriber2 = new TopicReceiver("testTopic2", JMSAcknowledgeMode.AUTO_ACKNOWLEDGE, configurationReader,
+                    "sub2");
+
+            //register listeners
+            subscriber.registerMessageListener();
+            subscriber2.registerMessageListener();
+
+            //send 10 messages
+            sender = new TopicSender("testTopic2", JMSAcknowledgeMode.AUTO_ACKNOWLEDGE, configurationReader);
+            sender.sendMessages(10, "test message");
+
+            TimeUnit.SECONDS.sleep(2L);
+            Assert.assertEquals(subscriber.receivedMessageCount(), 10, "All the published topics not received");
+            Assert.assertEquals(subscriber2.receivedMessageCount(), 10, "All the published topics not received");
+
+            //disconnect subscriber2
+            subscriber2.closeReceiver();
+
+            //send another 10 messages
+            sender.sendMessages(10, "test message");
+            subscriber2 = new TopicReceiver("testTopic2", JMSAcknowledgeMode.AUTO_ACKNOWLEDGE, configurationReader,
+                    "sub2");
+            subscriber2.registerMessageListener();
+            TimeUnit.SECONDS.sleep(2L);
```

Comment:
We don't need to make the sleeping time depend on the number of messages. Publishing doesn't need to be finished by the time the subscribers start listening. It's engoug if a message delivery can be started. Thus let's make this a constant which is the same as the previous.